Sources/PubNub/Networking/Routers/SubscribeRouter.swift

```diff
@@ -106,6 +106,37 @@ struct SubscribeRouter: HTTPRouter {
 struct SubscribeResponseDecoder: ResponseDecoder {
   typealias Payload = SubscriptionResponsePayload
 
+  func decode(response: EndpointResponse<Data>) -> Result<EndpointResponse<Payload>, Error> {
+    do {
+      let decodedPayload = try Constant.jsonDecoder.decode(Payload.self, from: response.payload)
+
+      let decodedResponse = EndpointResponse<Payload>(router: response.router,
+                                                      request: response.request,
+                                                      response: response.response,
+                                                      data: response.data,
+                                                      payload: decodedPayload)
+
+      return .success(decodedResponse)
+    } catch {
+      // Atempt to parse out the timetoken protion of the payload to push ahead the subscribe loop
+      if response.payload.count >= 37 {
```

Comment:
Are we guaranteed this length to be fixed?

Reply:
From what I've read JSON data is guaranteed to be ordered, and the goal of this code is to pull out the time payload from the response.  

The only variance would be whether the Timetoken or Region Int byte size were to change, or the system starts sending the time payload after the message portion, and neither of those are really controlled by the client so this is really just a better-effort attempt to keep the loop going. Worst case this fails, and then they're no worse off than they were previously (i.e. They get an error that they couldn't continue, and their loop stops)

Reply:
Ok thanks for the explanation. Sounds logical enough to use this method until we can't.